lib/cc/engine/analyzers/parser_base.rb

```diff
@@ -0,0 +1,13 @@
+module CC
+  module Engine
+    module Analyzers
+      class ParserBase
+        protected
```

Comment:
I think this can be `private`. Ruby's method access is an odd-ball:

- private - cannot be called with a target (even `self.` will raise)
- protected - can only be called with a target if you're the same class or a subclass

  This is mostly (only?) useful for otherwise private attributes that need to invoked with a target as part of an equality check:

  ```rb
  class Foo
    def ==(other)
      private_thing == other.private_thing
    end
  end
  ```

  This won't work if `#private_thing` is actually `private`, so it would be `protected`.